gh-89488: Add warning about `Py_BuildValue("p")` needing exact `int` (#135610)
python/cpython: Doc/c-api/arg.rst

   ``p`` (:class:`bool`) [int]
      Convert a C :c:expr:`int` to a Python :class:`bool` object.

>       Be aware that this format requires an ``int`` argument.
>       Unlike most other contexts in C, variadic arguments are not coerced to
>       a suitable type automatically.
>       You can convert another type (for example, a pointer or a float) to a
>       suitable ``int`` value using ``(x) ? 1 : 0`` or ``!!x``.

Reviewer: Would it make sense to put this inside a `.. note::` directive, like the `.. note:: This format does not accept bytes-like objects` above?

```suggestion
      .. note::
         This format requires an ``int`` argument.
         Unlike most other contexts in C, variadic arguments are not coerced to
         a suitable type automatically.
         You can convert another type (for example, a pointer or a float) to a
         suitable ``int`` value using ``(x) ? 1 : 0`` or ``!!x``.
```

Author: I wouldn't mind, but, I don't think this needs to be called out so prominently. An extra paragraph is already pretty noticeable.
